Add meta tensor support for squeeze

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #74376
* #74317
* __->__ #74314
* #74313
* #74375

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D34931336](https://our.internmc.facebook.com/intern/diff/D34931336)

ezyang committed:
Add meta tensor support for squeeze

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34931336).

Chillee reviewed (approved):
Kinda surprised this works, but given it (presumably) does, LGTM!

ezyang committed:
Update on "Add meta tensor support for squeeze"


Signed-off-by: Edward Z. Yang <ezyangfb.com>

Differential Revision: [D34931336](https://our.internmc.facebook.com/intern/diff/D34931336)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34931336).

ezyang committed:
Update on "Add meta tensor support for squeeze"


Signed-off-by: Edward Z. Yang <ezyangfb.com>

Differential Revision: [D34931336](https://our.internmc.facebook.com/intern/diff/D34931336)

[ghstack-poisoned]

ezyang commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -4162,15 +4162,15 @@
   device_guard: False
   dispatch:
-    CPU, CUDA: squeeze
+    CPU, CUDA, Meta: squeeze
```
let's hurry up and merge those PRs then :)

> They actually were previously, and a PR a month or two ago replaced CompositeExplicitAutograd with CPU, CUDA, regressing the coverage for the meta backend (too bad that the test suite didn't catch it :(

I have a new plan for testing based on OpInfos, hopefully will catch this

bdhirsh replied:
nice - landing here (hopefully today): D35016023
